sktime/classifiers/shapelet_based/mrseql/mrseql.pyx

```diff
@@ -0,0 +1,439 @@
+        if isinstance(X, pd.DataFrame):
+            if X.shape[1] > 1:
+                raise TypeError("Mr-SEQL cannot handle multivariate problems yet")
+            elif isinstance(X.iloc[0, 0], pd.Series):
+                X = np.asarray([a.values for a in X.iloc[:, 0]])
+            else:
+                raise TypeError("Input should either be a 2d numpy array, or a pandas dataframe with a single column of Series objects.")
+        return X
+
+    def fit(self, X, y, input_checks=True):
+
+        X = self.__X_check(X)
+
+        # transform time series to multiple symbolic representations
+        mr_seqs = self.__transform_time_series(X)
+
+        self.classes_ = np.unique(y) #because sklearn also uses np.unique
+
+        if self.__is_multiclass(): #one versus all
+            self.seql_models = self.__fit_multiclass_problem(mr_seqs, y)
+        else:
+            temp_labels = self.__to_tmp_labels(y)
+            self.seql_models = self.__fit_binary_problem(mr_seqs, temp_labels)
+
+        # if seql is being used to select features
+        # first computing the feature vectors
+        # then fit the new data to a logistic regression model
+        if self.seql_mode == 'fs':
+            train_x = self.__to_feature_space(mr_seqs)
+            self.clf = LogisticRegression(solver='newton-cg',multi_class = 'multinomial', class_weight='balanced').fit(train_x, y)
+            self.classes_ = self.clf.classes_ # shouldn't matter
+
+    def __compute_proba(self, score):
+        # if score < -8000:
+        #    return 0
+        # else:
+        return 1.0 / (1.0 + np.exp(-score))
+
+    def predict_proba(self, X, input_checks=True):
+        if input_checks:
+            X = self.__X_check(X)
+        mr_seqs = self.__transform_time_series(X)
+
+        if self.seql_mode == 'fs':
+            test_x = self.__to_feature_space(mr_seqs)
+            return self.clf.predict_proba(test_x) # TODO: Check if sklearn clf store labels in the same order
+
+        else:
+            scores = np.zeros((len(X), len(self.classes_)))
+            for rep, model in zip(mr_seqs, self.seql_models):
+                    for c,seq in enumerate(rep):
+                        scores[c] = scores[c] + model.classify(seq)
+
+            proba = self.__compute_proba(scores)
+            proba /= proba.sum(axis=1).reshape((proba.shape[0], -1)) # https://github.com/scikit-learn/scikit-learn/blob/bf24c7e3d6d768dddbfad3c26bb3f23bc82c0a18/sklearn/linear_model/_base.py#L300
+
+            return proba
+
+    def predict(self, X, input_checks=True):
+        if input_checks:
+            X = self.__X_check(X)
+        proba = self.predict_proba(X, False)
+        return np.array([self.classes_[np.argmax(prob)] for prob in proba])
+
+
+    def get_configuration(self):
+        return self.config 
+
+    # map sax features on time series
+    # return ndarray of (number of classes, length of time series)        
+    def map_sax_model(self, ts):
```

Comment:
Since this is a public function, could you add some docstrings explaining what it does or make it private